### PR TITLE
fix proto nitrate negative temperature + temperature fix for other reactions

### PR DESCRIPTION
**Pull request by Ghilker**
## About The Pull Request
Fixed negative energy that allowed proto nitrate to go negative temperature,
fixed other reactions from possible negative temperature

## Why It's Good For The Game
fex

## Changelog
:cl:
fix: fixed some atmos reaction from having negative temperature
/:cl:

### CODE REVIEW

**Comment by LemonInTheDark** [comment]
The issue here was most likely the large multiplier zauker has, but this is good as a safety check.

File: code/modules/atmospherics/gasmixtures/reactions.dm [hunk 1]
```diff
@@ -958,7 +958,7 @@ nobiliumsuppression = INFINITY
 
 		var/new_heat_capacity = air.heat_capacity()
 		if(new_heat_capacity > MINIMUM_HEAT_CAPACITY)
-			air.temperature = (temperature * old_heat_capacity + energy_released) / new_heat_capacity
+			air.temperature = max(((temperature * old_heat_capacity + energy_released) / new_heat_capacity), TCMB)
```
Comment (LemonInTheDark) [inline code/modules/atmospherics/gasmixtures/reactions.dm:961]: You don't need the second () there, in fact it clutters the code a bit
```suggestion
			air.temperature = max((temperature * old_heat_capacity + energy_released) / new_heat_capacity, TCMB)
```
Most all of the maxes have this issue

**Comment by LemonInTheDark** [comment]
You should clean up the rest too

**Comment by Ghilker** [comment]
Ah right will do

File: code/modules/atmospherics/gasmixtures/reactions.dm [hunk 2]
```diff
@@ -999,7 +999,7 @@ nobiliumsuppression = INFINITY
 	if(energy_released)
 		var/new_heat_capacity = air.heat_capacity()
 		if(new_heat_capacity > MINIMUM_HEAT_CAPACITY)
-			air.temperature = (temperature * old_heat_capacity + energy_released) / new_heat_capacity
+			air.temperature = max(((temperature * old_heat_capacity + energy_released) / new_heat_capacity), TCMB)
```
Comment (Ghilker) [inline code/modules/atmospherics/gasmixtures/reactions.dm:1002]:
```suggestion
			air.temperature = max((temperature * old_heat_capacity + energy_released) / new_heat_capacity, TCMB)
```

File: code/modules/atmospherics/gasmixtures/reactions.dm [hunk 3]
```diff
@@ -1032,7 +1032,7 @@ nobiliumsuppression = INFINITY
...
```
Comment (Ghilker) [inline code/modules/atmospherics/gasmixtures/reactions.dm:1035]:
```suggestion
			air.temperature = max((temperature * old_heat_capacity + energy_released) / new_heat_capacity, TCMB)
```

File: code/modules/atmospherics/gasmixtures/reactions.dm [hunk 4]
```diff
@@ -1056,11 +1056,11 @@ nobiliumsuppression = INFINITY
 		return NO_REACTION
 	cached_gases[/datum/gas/hydrogen][MOLES] -= produced_amount
 	cached_gases[/datum/gas/proto_nitrate][MOLES] += produced_amount * 0.5
-	energy_released -= produced_amount * 2500
+	energy_released = produced_amount * 2500
 	if(energy_released)
 		var/new_heat_capacity = air.heat_capacity()
 		if(new_heat_capacity > MINIMUM_HEAT_CAPACITY)
-			air.temperature = (temperature * old_heat_capacity + energy_released) / new_heat_capacity
+			air.temperature = max(((temperature * old_heat_capacity - energy_released) / new_heat_capacity), TCMB)
```
Comment (Ghilker) [inline code/modules/atmospherics/gasmixtures/reactions.dm:1063]:
```suggestion
			air.temperature = max((temperature * old_heat_capacity - energy_released) / new_heat_capacity, TCMB)
```

File: code/modules/atmospherics/gasmixtures/reactions.dm [hunk 5]
```diff
@@ -1116,5 +1116,5 @@ nobiliumsuppression = INFINITY
 	if(energy_released)
 		var/new_heat_capacity = air.heat_capacity()
 		if(new_heat_capacity > MINIMUM_HEAT_CAPACITY)
-			air.temperature = (temperature * old_heat_capacity + energy_released) / new_heat_capacity
+			air.temperature = max(((temperature * old_heat_capacity + energy_released) / new_heat_capacity), TCMB)
```
Comment (Ghilker) [inline code/modules/atmospherics/gasmixtures/reactions.dm:1119]:
```suggestion
			air.temperature = max((temperature * old_heat_capacity + energy_released) / new_heat_capacity, TCMB)
```

**Pull request merged**